pkg/middleware/scope.go

```diff
@@ -6,26 +6,26 @@ import (
 
 	"github.com/justinas/alice"
 	middlewareapi "github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/middleware"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
 )
 
 type scopeKey string
 
-// requestScopeKey uses a typed string to reduce likelihood of clasing
+// requestScopeKey uses a typed string to reduce likelihood of clashing
 // with other context keys
 const requestScopeKey scopeKey = "request-scope"
 
-func NewScope() alice.Constructor {
-	return addScope
-}
-
-// addScope injects a new request scope into the request context.
-func addScope(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		scope := &middlewareapi.RequestScope{}
-		contextWithScope := context.WithValue(req.Context(), requestScopeKey, scope)
-		requestWithScope := req.WithContext(contextWithScope)
-		next.ServeHTTP(rw, requestWithScope)
-	})
+func NewScope(opts *options.Options) alice.Constructor {
```

Comment:
I would prefer we use our own configuration struct here rather than relying on the complete Options struct. Ideally we decouple as many parts of the options struct from other parts of the codebase as possible. So for example, other middelwares take in a part of the config, eg a list of headers. 

We could just have it as a single bool for now or create a struct to take the options, not sure what else might go into the scope as global config though 🤔

Reply:
Good point, I'll flip to the single bool.  I'm not sure what other options we have (now or in the future) besides this one where we would track specifically related to `requests` and are best stored on their context vs the rest of the options that configure stuff like the OAuthProxy struct itself.